### ISSUE TITLE
Memory leak with DB insert multiple records

### ISSUE THREAD

**Issue by sfilzek** [laravel/framework#35319]
- Laravel Version: 8.15.0
- PHP Version: 7.4.10
- Database Driver & Version: MySql 5.7.24 

### Description:

Using DB::insert() with multiple records by passing an array of arrays (as per: https://laravel.com/docs/8.x/queries#inserts) seems to leak memory.

### Steps To Reproduce:

```php
    public function runBatch()
    {   
        $tempTable = 'csv_test_batch';

        Schema::create($tempTable, function (Blueprint $table) {
            $table->string('field_1');
            $table->string('field_2');
            $table->string('field_3');
            $table->string('field_4');
            $table->string('field_5');
        });
        
        $recordCount = 0;
        while ($recordCount < 50000) {
            $chunkSize = 5000;
            $data = [];
            while ($chunkSize-- > 0) {
                $data[] = [
                    'field_1' => 'Some data 1',
                    'field_2' => 'Some data 2',
                    'field_3' => 'Some data 3',
                    'field_4' => 'Some data 4',
                    'field_5' => 'Some data 5',
                ];
                $recordCount++;
            }
            DB::table($tempTable)->insert($data);
            info("Batch Insert: {$recordCount} records, memory: ". number_format(memory_get_usage()));
        }
    }
```

Running this shows the increasing memory usage:

```
[2020-11-22 07:44:11] local.INFO: Batch Insert: 5000 records, memory: 30,458,104  
[2020-11-22 07:44:12] local.INFO: Batch Insert: 10000 records, memory: 31,826,240  
[2020-11-22 07:44:12] local.INFO: Batch Insert: 15000 records, memory: 32,965,304  
[2020-11-22 07:44:12] local.INFO: Batch Insert: 20000 records, memory: 34,104,368  
[2020-11-22 07:44:12] local.INFO: Batch Insert: 25000 records, memory: 35,243,432  
[2020-11-22 07:44:12] local.INFO: Batch Insert: 30000 records, memory: 36,382,496  
[2020-11-22 07:44:12] local.INFO: Batch Insert: 35000 records, memory: 37,521,560  
[2020-11-22 07:44:12] local.INFO: Batch Insert: 40000 records, memory: 38,660,944  
[2020-11-22 07:44:12] local.INFO: Batch Insert: 45000 records, memory: 39,800,008  
[2020-11-22 07:44:12] local.INFO: Batch Insert: 50000 records, memory: 40,939,392  
````

Running the same set of queries as single inserts (while much slower) does not seem to cause a memory leak:

````php
    public static function runSingle()
    {   
        $tempTable = 'csv_test_single';

        Schema::create($tempTable, function (Blueprint $table) {
            $table->string('field_1');
            $table->string('field_2');
            $table->string('field_3');
            $table->string('field_4');
            $table->string('field_5');
        });

        $recordCount = 0;
        while ($recordCount < 50000) {
            $chunkSize = 5000;
            while ($chunkSize-- > 0) {
                $data = [
                    'field_1' => 'Some data 1',
                    'field_2' => 'Some data 2',
                    'field_3' => 'Some data 3',
                    'field_4' => 'Some data 4',
                    'field_5' => 'Some data 5',
                ];
                $recordCount++;
                DB::table($tempTable)->insert($data);
            }
            info("Single Insert: {$recordCount} records, memory: ". number_format(memory_get_usage()));
        }
    }
````

The result is:

````
[2020-11-22 07:44:39] local.INFO: Single Insert: 5000 records, memory: 29,503,992  
[2020-11-22 07:44:53] local.INFO: Single Insert: 10000 records, memory: 29,504,184  
[2020-11-22 07:45:07] local.INFO: Single Insert: 15000 records, memory: 29,504,376  
[2020-11-22 07:45:22] local.INFO: Single Insert: 20000 records, memory: 29,504,568  
[2020-11-22 07:45:36] local.INFO: Single Insert: 25000 records, memory: 29,504,760  
[2020-11-22 07:45:52] local.INFO: Single Insert: 30000 records, memory: 29,504,952  
[2020-11-22 07:46:06] local.INFO: Single Insert: 35000 records, memory: 29,505,144  
[2020-11-22 07:46:21] local.INFO: Single Insert: 40000 records, memory: 29,505,976  
[2020-11-22 07:46:35] local.INFO: Single Insert: 45000 records, memory: 29,506,168  
[2020-11-22 07:46:49] local.INFO: Single Insert: 50000 records, memory: 29,506,360  
````

I've tried adding DB::disableQueryLog() just in case something was enabling it, but that does not seem to make a difference.

**Comment by crynobone** [comment]
You might refer to #35095

**Comment by sfilzek** [comment]
This issue was happening on a new clean Laravel 8.15.0 install.

Thanks for the heads-up #crynobone,  As per https://github.com/laravel/framework/issues/35095#issuecomment-722398532 I removed facade/ignition and the memory leak has stopped.

I've tried the same thing on my actual code base and I can confirm that removing Ignition has stopped the leak there also. So this seems to be an issue with Ignition when using inserting arrays of arrays. Strange, why would Ignition care if we performed 50K single inserts vs 10 inserts of 5K records...

I'll close here and open an issue on facade/ignition, thanks!
